R/lvl0_mergeDat.R

```diff
@@ -112,6 +112,22 @@ lvl0_mergeDat = function(UCD_output, EU_data, PSI_costs, altCosts, CHN_trucks, G
   PSI_c[, c("variable", "unit") := list("Capital costs (purchase)", "2005$/veh/yr")]
   PSI_c = merge(PSI_c, unique(AM[, c("iso", "year")]), by = "year", allow.cartesian=TRUE)
 
+  ## load the UCD based purchase costs
+  UCD_c = copy(UCD_output$UCD_cost)
+  ## find the purchase costs of liquid and NG technologies for non-EU countries (for EU, data comes from PSI)
+  purchCost = UCD_c[!iso %in% eu_iso & variable %in% "Capital costs (purchase)" & UCD_technology %in% c("Liquids", "NG")][, c("iso", "UCD_technology", "year", "value", "vehicle_type")]
+  setnames(purchCost, old="value", new="valUCD")
+  setnames(purchCost, old="UCD_technology", new="technology")
+  ## merge the data from UCD with the PSI values as they contain the corresponding markups for each tech/veh type
+  purchCost=merge(PSI_c, purchCost, by = c("year", "technology", "iso", "vehicle_type"))
+  purchCost[, tot_purchasecost := valUCD]
+  purchCost[, valUCD := NULL]
+
+  ## remove the "extra vehicle types" that have purchase cost associated but not all other costs - they are all in non_eu countries- as they are not in the original demand trends
```

Comment:
What do you mean by that?:)

Reply:
it is an artifact of the merging across data tables... some purchase costs are created for all vehicle types, to cover all vehicle options. However not every region/iso has all the vehicles, and the cross-join across data tables leads to creating fake purchase costs for regions that are not supposed to have some vehicle types. This line makes sure they are removed. When we restructure the code, we have to make sure to avoid such complications, linked to the loading of different dfs with different detail and regional aggregation.

Reply:
Totally! These different size categories across regions are a constant source of nuisance.